ocp_resources/pipeline.py

```diff
@@ -31,12 +24,7 @@ def __init__(
             tasks have completed in parallel.
             spec section can't be empty. It requires at least one optional field.
         """
-        super().__init__(
-            name=name,
-            namespace=namespace,
-            client=client,
-            **kwargs,
-        )
+        super().__init__(**kwargs)
```

Comment:
Can these resources be generated by utilizing https://github.com/RedHatQE/openshift-python-wrapper/tree/main/class_generator please?

Reply:
oc explain doesn't show spec details . I have a bug too for that

Reply:
Please try again, we do not use oc explain anymore to generate resources

Reply:
I did give it a try :+1: 
1. I first updated the schema to get tekton resource json.

$ cat class_generator/schema/pipelinerun.json
{
  "type": "object",
  "x-kubernetes-group-version-kind": [
    {
      "group": "tekton.dev",
      "kind": "PipelineRun",
      "version": "v1beta1"
    }
  ],
  "$schema": "http://json-schema.org/schema#"

$ cat class_generator/schema/pipeline.json
{
  "type": "object",
  "x-kubernetes-group-version-kind": [
    {
      "group": "tekton.dev",
      "kind": "Pipeline",
      "version": "v1beta1"
    }
  ],
  "$schema": "http://json-schema.org/schema#"


2. I tried with working kind which is DNS in my setup just to be sure that setup is correct, it worked

3. When trying for tekton



$ class-generator --kind PipelineRun
2024-10-24T10:39:42.859879 class_generator INFO Generating class for PipelineRun
Traceback (most recent call last):
  File "/home/gkapoor/.local/bin/class-generator", line 8, in <module>
    sys.exit(main())
  File "/home/gkapoor/.local/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/class_generator/class_generator.py", line 721, in main
    class_generator(**_kwargs)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/class_generator/class_generator.py", line 569, in class_generator
    resources = parse_explain(kind=kind)
  File "/home/gkapoor/.local/lib/python3.9/site-packages/class_generator/class_generator.py", line 476, in parse_explain
    schema_properties: Dict[str, Any] = _kind_schema["properties"]
KeyError: 'properties'

Reply:
Thanks